### ISSUE TITLE
(1) proj.pt for 10k PCA components (2) std.pt has negative values

### ISSUE THREAD

**Issue by Ab-34** [snap-research/weights2weights#10]
Hi, thank you very much for you code.

I was trying to sample from the w2w space. However, the proj is a 64974x1000 with the top 1000 pca values, but my set-up includes the top 10k pca values. **So could you please provide the 64974x10k matrix?**

As an alternative, I was also trying to use the mean and std files that have been provided to sample from the top 10k components, **but I found negative values in the std.pt file, so I wanted to ask about that too.**

Hoping to hear from you,
Thank you

**Comment by avdravid** [comment]
Hi! Thank you for raising the issue. What I suggest is going to: https://huggingface.co/snap-research/weights2weights/tree/main/weights_datasets/identities. There we have a file called ``all_weights.pt`` which is 64974x~100,000 dimensional tensor which is the full weight vector for each identity. Next, you can run: 

a = torch.load("all_weights.pt")
mean = torch.mean(a, dim=0)
std = torch.std(a, dim=0)
scaled_data = (a - mean)/std
u,s,v = torch.svd(scaled_data)

If you get out-of-memory issues when running SVD on large tensors, you can also use torch.svd_lowrank. Going above 20,000 principal components becomes computationally infeasible.

The negative values in ``std.pt`` is due to numerical precision reasons. But I did not notice that it caused any issues.
